eucadeploy/plugins/debugger/eucalyptus_sosreports.py

```diff
@@ -0,0 +1,150 @@
+                else:
+                    self.warning(host + ':Package not installed - '
+                                + package + '; Installing ' + package)
+                    with hide('everything'):
+                        yum_output = self.run_command_on_host('yum install '
+                                     + ' --assumeyes --quiet --nogpgcheck ' + package,
+                                     host=host)
+                    if not yum_output:
+                        self.success(host + ':Package installed - ' + package)
+                    else:
+                        self.failure(host + ':Package failed to install - ' + package)
+
+    def _grab_sosreports(self, all_hosts):
+        """
+        Create local directory
+        for downloaded sosreports
+        """
+        timestamp = datetime.strftime(datetime.now(), '%Y%m%d-%H%M%S') 
+        directory = 'sosreport-' + timestamp
+        message = 'Creating directory ' + directory + " on localhost"
+        self.info(message)
+        with hide('everything'):
+            mkdir_output = self.create_localdir(directory)
+        if re.search('created', mkdir_output):
+            self.success('localhost: Directory ' + directory
+                         + ' created successfully')
+        else:
+            self.failure('localhost: Directory ' + directory
+                         + ' creation failed')
+        """
+        Remove old sosreports are around in /tmp
+        """
+        with hide('everything'):
+            rm_output = self.run_command_on_hosts('rm -rf /tmp/sosreport*',
+                                                  hosts=all_hosts)
+        for host in all_hosts:
+            if not rm_output[host]:
+                self.success(host + ':old sosreports successfully removed')
+            else:
+                self.failure(host + ':old sosreports failed to be removed')
+
+        with hide('everything'):
+            """
+            Execute sosreport on all hosts
+            """
+            sosreport_output = self.execute_sosreports_on_hosts(hosts=all_hosts)
+        for host in all_hosts:
+            """
+            Confirm sosreport ran successfully;
+            Download sosreport to local client
+            """
+            hostname = host.replace(".", '')
+            sosfile = 'sosreport-' + hostname
+            for output in sosreport_output[host].split('\n'):
+                sosreport_file = re.search(sosfile, output, re.I)
+                if sosreport_file:
+                    self.success(host + ':sosreport finished - '
+                                + output)
+                    remote_path = output.strip()
+                    local_file = remote_path.split('/')[2]
+                    format_host = host.replace(".", "_")
+                    local_path = directory + "/" + format_host + "-" + local_file
+                    with hide('everything'):
+                        cp_output = self.get_file_on_host(remote_path,
```

Comment:
What is actually being returned here? Will it ever be None? My concern is that we could end up with false positives if the check below always passes.